ios/ElectrodeApiImpl/APIImpls/Navigation/ENNavigationDelegate.swift

```diff
@@ -222,18 +222,29 @@ import UIKit
         }
         button.isEnabled = !(navigationButton.disabled ?? false)
         button.stringTag = navigationButton.id
+        button.currViewController = vc
         return button
     }
 
     @objc func clickButtonWithButtonId(_ sender: ENBarButtonItem) {
         if let stringTag = sender.stringTag {
             ENNavigationAPIImpl.shared.navigationAPI.events.emitEventOnNavButtonClick(buttonId: stringTag)
+            var viewId = "NOT_SET"
+            if let vc = sender.currViewController as? MiniAppNavViewController {
+                viewId = vc.viewIdentifier
+            }
+            ENNavigationAPIImpl.shared.navigationAPI.events.emitEventNavEvent(eventData: NavEventData(eventType: NavEventType.BUTTON_CLICK.rawValue, viewId: viewId, jsonPayload: getButtonJsonPayload(buttonId: stringTag)))
```

Comment:
this logic is growing and is repeated code, could move to helper function

Reply:
👍